Userland/Libraries/LibWeb/PixelUnits.h

```diff
@@ -50,151 +52,75 @@ constexpr DevicePixels operator%(DevicePixels left, T right) { return left.value
-    {
-        return to_double();
-    }
+    int operator<=>(CSSPixels const& other) const;
 
-    constexpr CSSPixels& operator++()
-    {
-        this->m_value += 1;
-        return *this;
-    }
-    constexpr CSSPixels operator++(int)
-    {
-        CSSPixels ret = this->m_value;
-        this->m_value += 1;
-        return ret;
-    }
-    constexpr CSSPixels& operator--()
-    {
-        this->m_value -= 1;
-        return *this;
-    }
-    constexpr CSSPixels operator--(int)
-    {
-        CSSPixels ret = this->m_value;
-        this->m_value -= 1;
-        return ret;
-    }
+    CSSPixels operator+() const;
+    CSSPixels operator-() const;
 
-    constexpr int operator<=>(CSSPixels const& other) const
-    {
-        return this->m_value > other.m_value ? 1 : this->m_value < other.m_value ? -1
-                                                                                 : 0;
-    }
+    CSSPixels operator+(CSSPixels const& other) const;
+    CSSPixels operator-(CSSPixels const& other) const;
+    CSSPixels operator*(CSSPixels const& other) const;
+    CSSPixels operator/(CSSPixels const& other) const;
 
-    constexpr CSSPixels operator+(CSSPixels const& other) const
-    {
-        return this->m_value + other.m_value;
-    }
+    CSSPixels& operator+=(CSSPixels const& other);
+    CSSPixels& operator-=(CSSPixels const& other);
+    CSSPixels& operator*=(CSSPixels const& other);
+    CSSPixels& operator/=(CSSPixels const& other);
 
-    constexpr CSSPixels operator-(CSSPixels const& other) const
-    {
-        return this->m_value - other.m_value;
-    }
+    CSSPixels abs() const;
 
-    constexpr CSSPixels operator+() const
-    {
-        return +this->m_value;
-    }
-
-    constexpr CSSPixels operator-() const
-    {
-        return -this->m_value;
-    }
-
-    constexpr CSSPixels operator*(CSSPixels const& other) const
-    {
-        return this->m_value * other.m_value;
-    }
-
-    constexpr CSSPixels operator/(CSSPixels const& other) const
-    {
-        return this->m_value / other.m_value;
-    }
-
-    constexpr CSSPixels& operator+=(CSSPixels const& other)
-    {
-        this->m_value += other.m_value;
-        return *this;
-    }
-
-    constexpr CSSPixels& operator-=(CSSPixels const& other)
-    {
-        this->m_value -= other.m_value;
-        return *this;
-    }
-
-    constexpr CSSPixels& operator*=(CSSPixels const& other)
-    {
-        this->m_value *= other.m_value;
-        return *this;
-    }
-
-    constexpr CSSPixels& operator/=(CSSPixels const& other)
-    {
-        this->m_value /= other.m_value;
-        return *this;
-    }
+    static float epsilon();
 
 private:
-    double m_value {};
+    int m_value { 0 };
 };
 
 template<Arithmetic T>
-constexpr bool operator==(CSSPixels left, T right) { return left.to_double() == right; }
+constexpr bool operator==(CSSPixels left, T right) { return left.to_double() == static_cast<double>(right); }
```

Comment:
Do we need these explicit casts to `double`? Implicit conversions should work just like they did before changing the underlying implementation of `CSSPixels`.

Reply:
these operators that always cast to double were hacks and I replaced them with explicit operator definitions for float/double/int values.